controllers/k8ssandra/datacenters.go

```diff
@@ -78,14 +90,38 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 
 		actualDc := &cassdcapi.CassandraDatacenter{}
 
-		remoteClient, err := r.ClientCache.GetRemoteClient(dcTemplate.K8sContext)
+		if recResult := r.reconcileSeedsEndpoints(ctx, desiredDc, seeds, remoteClient, logger); recResult.Completed() {
+			return recResult, actualDcs
+		}
+
+		desiredConfig, err := utils.UnmarshalToMap(desiredDc.Spec.Config)
 		if err != nil {
-			logger.Error(err, "Failed to get remote client")
 			return result.Error(err), actualDcs
 		}
 
-		if recResult := r.reconcileSeedsEndpoints(ctx, desiredDc, seeds, remoteClient, logger); recResult.Completed() {
-			return recResult, actualDcs
+		if dcTemplate.Stargate != nil {
```

Comment:
I am wondering if this reconciliation logic shouldn't be done by the Stargate controller:

* The k8ssandra controller remains responsible for creating the `Stargate` resource and filling it with the required specs, including which secrets should be used for encryption.
* Then the Stargate controller picks up the Stargate resource and "merges" those secrets with whatever is specified in `CassandraConfigMapRef` to create a new merged configmap and use it.

I believe that this way it would be easier for standalone Stargate resources to be able to be reconciled correctly.